drafts/Stormless3.md

```diff
@@ -0,0 +1,322 @@
+
+In the [Episode Two][ep-two], we added more actions: one native Lambda to record user info to DynamoDB, and another one from StackStorm Exchange to make a call to ActiveCampaign CRM system. You learned more of `serverless.yml` syntax and practiced developer workflow working with individual functions. 
+
+[ep-one]:https://medium.com/@dzimine/tutorial-building-a-community-on-boarding-app-with-serverless-stepfunctions-and-stackstorm-b2f7cf2cc419
+
+[ep-two]: https://medium.com/@dzimine/building-community-sign-up-app-with-serverless-stepfunctions-and-stackstorm-exchange-episode-2-b1efeb1b9bd6
+[st2-ex]: https://exchange.stackstorm.org 
+
+In this episode, I'll show how to use AWS StepFunction to wire the actions into a workflow.
+
+The complete code for this episode is on Github at [3-add-stepfunction](https://github.com/dzimine/slack-signup-serverless-stormless/tree/DZ/3-add-stepfunction) branch. 
+
+
+#### Wire functions together with StepFunction
+
+Now that the building blocks are lined up, time to wire them together. I'll use [`serverless-step-functions` plugin](https://github.com/horike37/serverless-step-functions) from Serverless Champion ... 
+
+https://twitter.com/goserverless/status/941777242685767680
+
+Install the plugin: 
+```
+npm install --save-dev serverless-step-functions
+```
+
+Add the plugin to your serverless.yml file
+
+```
+plugins:
+  - serverless-plugin-stackstorm
+  - serverless-step-functions
+```
+
+The Step Function definition will require my `accountID`; as this is a thing I don't like sharing with anyone, I put it to `env.yml`, which now look like this:
+
+```
+# env.yml
+# Don't commit to Github :)
+slack:
+  admin_token: "xoxs-111111111111-..."
+  organization: "your-team"
+active_campaign:
+  url: "https://YOUR-COMPANY.api-us1.com"
+  api_key: "1234567a9012a12345z12aa2aaa..."
+private:
+  accountId: "000000000000"
+```
+
+Back to `serverless.yml` with a couple of key touches there:
+
+```
+custom:
+  private: ${file(env.yml):private}
+  stage: ${opt:stage, self:provider.stage}
+  region: ${opt:region, self:provider.region}
+
+stepFunctions:
+  stateMachines:
+    signup:
+      events:
+        - http:
+            path: signup
+            method: POST
+            cors: true
+      definition: ${file(stepfunction.yml)}
+```
+
+In the `custom` block, I assigned `private` from namesake key in `env.yml`, I also defined variables for `stage` and `region` so that the values are picked from CLI options, if provided, or defaults to the current AWS settings.
+
+The `stepFunctions` block is here to define ... drumroll ... you have already guessed - StepFunctions. Mine is called"`signup`". The `events` section here looks, acts and servers just like `events` in function definition: it configures an API Gateway endpoint for invoking StepFunction from outside of AWS. We'll use it later in a web form. The definition takes. It can be written as YAML right here in `serverless.yml`, 
```

Comment:
called"\`signup\`"

I would remove the double quotes, and make sure you add a space between the two words.

Reply:
Remove "and servers"? Or maybe you meant "serves"?

Reply:
> The definition takes.

Incomplete thought I think